standards/core/rln-contract.md

```diff
@@ -0,0 +1,204 @@
+This means that one user with a `1` message per second rate limit (high-tier) generates up to `600 * 128 / 1024 = 75 KiB` of nullifier log data per 10-min epoch.
+
+#### Maximum total message limit / concurrently active memberships
+
+We may want to limit the total rate limit available to all users.
+The rationale is that the total network bandwidth is a limited resource.
+The total active rate limit should not exceed the network's real capabilities.
+
+There may be two ways to implement a global cap:
+- limit the total rate limits;
+- limit the total number of active memberships (e.g., at 10K memberships).
+
+#### Proportional pricing
+
+We suggest proportional pricing for simplicity.
+Proportional pricing means that if tier A offers N times the limit of tier B, then tier A is N times more expensive than tier B.
+
+An alternative approach could be bulk discounts for high-tier memberships.
+Discounts are more efficient but promote centralization.
+Finding the right trade-off remains subject for future work.
+
+#### Accepted tokens
+
+When choosing a token to accept, we consider the following criteria:
+
+- a stablecoin: USD-denominated pricing is simpler for users and for developers (avoiding an oracle);
+- popular, high liquidity;
+- preferably decentralized;
+- with a reasonably good track record w.r.t. censorship.
+
+Based on these criteria, we suggest accepting DAI for the initial deployment.
+Other tokens may be added in the future.
+
+#### Grace period and membership extension
+
+On the one hand, memberships must expire,
+otherwise a one-time payment would give a user the right to potentially infinite resource usage.
+On the other hand, the “soft” expiration suggested above is likely sufficient
+while balancing abuse prevention and ease of implementation:
+
+- although extending a membership doesn’t require a new deposit, keeping the original deposit locked up plus paying gas fees imposes cost on membership extension;
+- a legitimate user is unlikely to risk their membership being taken over at any moment by not renewing their membership.
+
+## Implementation Suggestions
+
+The current version of the contract (RLNv2) is deployed on Sepolia testnet ([source code](https://github.com/waku-org/waku-rlnv2-contract/blob/main/src/WakuRlnV2.sol)).
+
+## Security / Privacy Considerations
+
+Requesting a Merkle proof for one's own membership through a third-party RPC provider may endanger the requester's privacy.
+
+## Future work
+
+In the future version of RLN contract, the following research directions might be pursued:
+- deposit slashing for misbehaving users;
+- extracting revenue by e.g. only allowing partial deposit withdrawal;
+- incentivization for relaying nodes;
+- non-proportional pricing (e.g., high-tier discounts);
+- accepting other tokens;
+- making memberships transferable (e.g., in the form of an NFT);
+- establishing a free relaying market by economically linking demand with supply;
```

Comment:
I know it's not a list of promises but is that something we are actually considering? cc @jm-clius

Reply:
Also is this really relevant for RLN spec?

Reply:
TBH, I included this section largely because of my academic habits, as research papers normally inculde "future work" even if you don't plan to implement anything yourself, more as a guidance to other researchers who would want to build on top of your work. It may well be the case that we don't needed future work in a spec. Or, perhaps, we could include a few points that are most likely to get implemented in the foreseeable future (AFAICT, slashing is not one of those).